src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -346,15 +347,53 @@ def _create_kubernetes(resource_group_name, deployment_name, dns_name_prefix, na
     smc = get_mgmt_service_client(ResourceManagementClient)
     return smc.deployments.create_or_update(resource_group_name, deployment_name, properties)
 
-def acs_get_credentials(dns_prefix, location):
-    # TODO: once we get the right swagger in here, update this to actually pull location and dns_prefix
-    #acs_info = _get_acs_info(name, resource_group_name)
-    home = os.path.expanduser('~')
+def acs_get_credentials(name=None, resource_group_name=None, dns_prefix=None, location=None):
+    if not dns_prefix or not location:
+        acs_info = _get_acs_info(name, resource_group_name)
+
+        if not dns_prefix:
+            dns_prefix = acs_info.master_profile.dns_prefix # pylint: disable=no-member
+        if not location:
+            location = acs_info.location # pylint: disable=no-member
 
+    home = os.path.expanduser('~')
     path = os.path.join(home, '.kube', 'config')
+
+    path_candidate = path
+    ix = 0
+    while os.path.exists(path_candidate):
+        ix += 1
+        path_candidate = '{}-{}-{}'.format(path, name, ix)
+
     # TODO: this only works for public cloud, need other casing for national clouds
-    acs_client.SecureCopy('azureuser', '{}-k8s-masters.{}.cloudapp.azure.com'.format(dns_prefix, location),
-                          '.kube/config', path)
+    acs_client.SecureCopy('azureuser', '{}.{}.cloudapp.azure.com'.format(dns_prefix, location),
+                          '.kube/config', path_candidate)
+
+    # merge things
+    if path_candidate != path:
+        merge_kubernetes_configurations(path, path_candidate)
+
+def merge_kubernetes_configurations(existing_file, addition_file):
+    with open(existing_file) as stream:
+        try:
+            existing = yaml.load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+            return
```

Comment:
Since you catch this error, this would result in the command `az acs kubernetes get-credentials` returning an exit code of 0 even if it failed to update the .kube/config file and the user won't know where the credentials were saved to.

Suggestion:
- Don't catch YAMLErrors in `merge_kubernetes_configurations`.
- In `acs_get_credentials` put something like the following around the call to `merge_kubernetes_configurations`.
```
try:
    merge_kubernetes_configurations(path, path_candidate)
except yaml.YAMLError as exc:
    logger.warning('Failed to merge credentials to kube config file: {}'.format(exc))
    logger.warning('The credentials have been saved to {}'.format(path_candidate))
```

This way, if the merging fails, the user will still be provided with the path to the credentials.